### PR TITLE
initial draft of tooling initiatives doc

### PR DESCRIPTION
**Pull request by boneskull**
Not sure *where* to put this doc exactly, but here it is.

References: #59, #68

### CODE REVIEW

**Comment by boneskull** [comment]
cc's

@mhdawson @dshaw @bnb @cb1kenobi @devsnek @iarna @TheLarkInn @hansl @stefanpenner @zertosh

File: groups/tooling/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  - [Example 2: The Test Case](#example-2-the-test-case)
+  - [Notes](#notes)
+
+<!-- /TOC -->
+
+## Add Common Filesystem Operations to Core
+
+### Summary
+
+Two (2) filesystem operations should be added to Node.js Core:
```
Comment (stefanpenner) [inline groups/tooling/README.md]: big +1, its not uncommon to have 15-25+ `mkdirp` modules in node_modules, and about the same for `fs-extra`

**Comment by boneskull** [comment]
Would anyone like to merge?

**Comment by mhdawson** [comment]
Landed as b371e0ad786c2f4a0aee76952bc7d1e52bf159bc

**Comment by bcoe** [comment]
@mhdawson @boneskull @stefanpenner 🎉I'm excited about this. Having been working on Node command line tools in my spare time for years at this point (yargs, nyc, Istanbul), I'm used to my first action often being to `npm i mkdirp` and `npm i rimraf`.

I also have some grander thoughts around tooling as it relates to testing. Specifically, it would be nice to have source-map support (CC: @addaleax) which would simply most transpilation approaches, and make for better stack traces. Eventually I'd also love for us to find a lighter-weight way to expose [v8 test coverage](https://v8project.blogspot.com/2017/12/javascript-code-coverage.html). (CC: @schuay). tldr; test coverage is built into v8, we shouldn't need to use transpilation to collect it in Node.js.

@boneskull and I have done a tiny bit of hacking so far on `mkdirp`, it seems like some great low hanging fruit. Let's get that `common-file-ops.md` started, which we can reference as we begin to evangelize tooling improvements.

**Comment by cb1kenobi** [comment]
+1 for source-map support as I previous suggested: https://github.com/nodejs/user-feedback/issues/59#issuecomment-396332169.

**Pull request merged**